### PR TITLE
Fixes incorrect joining of paths

### LINKED ISSUE #9: Empty FQDN in clients list
```
 user 暭 curl http://engineering:8080/clients
[{"targets":[""],"labels":null}]
 ~/src/go 暭 
 user 暭 
```

with

```
 user 暭 bin/client --proxy-url=http://engineering:8080/ --log.level=debug --fqdn="engineering"
```

and 

```
 user  bin/proxy 
level=info ts=2017-10-03T09:10:19.78520441Z caller=proxy.go:104 msg=Listening address=:8080
```

### PR DESCRIPTION
**Pull request by Conorbro**
Closes #9

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -77,18 +78,22 @@ func (c *Coordinator) doPush(resp *http.Response, origRequest *http.Request, cli
 	deadline, _ := origRequest.Context().Deadline()
 	resp.Header.Set("X-Prometheus-Scrape-Timeout", fmt.Sprintf("%f", float64(time.Until(deadline))/1e9))
 
-	u, _ := url.Parse(*proxyURL + "/push")
+	url, err := url.Parse(*proxyURL)
+	if err != nil {
+		return err
+	}
+	url.Path = path.Join(url.Path, "/post")
```
Comment (brian-brazil) [inline client/client.go]: Path is for files, not for urls. I think you want ResolveReference

**Comment by Conorbro** [comment]
@brian-brazil

File: client/client.go [hunk 2]
```diff
@@ -97,7 +105,21 @@ func (c *Coordinator) doPush(resp *http.Response, origRequest *http.Request, cli
...
+	base, err := url.Parse(*proxyURL)
+	if err != nil {
+		return
+	}
+	u, err := url.Parse("/poll")
+	if err != nil {
+		return
+	}
+	url := base.ResolveReference(u)
+	fmt.Println(url)
```
Comment (brian-brazil) [inline client/client.go]: Debug code?

**Comment by Conorbro** [comment]
@brian-brazil

File: client/client.go [hunk 3]
```diff
@@ -97,7 +105,20 @@ func (c *Coordinator) doPush(resp *http.Response, origRequest *http.Request, cli
...
+	if err != nil {
+		return
+	}
+	u, err := url.Parse("/poll")
+	if err != nil {
+		return
+	}
+	url := base.ResolveReference(u)
+	if err != nil {
+		level.Error(c.logger).Log("msg", "Error parsing url:", "err", err)
```
Comment (brian-brazil) [inline client/client.go]: This doesn't seem to be in the right place

**Comment by Conorbro** [comment]
@brian-brazil

**Pull request merged**